test/unit/util/conf/test_base_config_loader.py

```diff
@@ -1,10 +1,62 @@
-from app.util.conf.base_config_loader import BaseConfigLoader
+from genty import genty, genty_dataset
+
+from app.util.conf.base_config_loader import BaseConfigLoader, InvalidConfigError
 from app.util.conf.configuration import Configuration
 from test.framework.base_unit_test_case import BaseUnitTestCase
 
 
+class _FakeConfigLoader(BaseConfigLoader):
+    def _get_config_file_whitelisted_keys(self):
+        return ['some_bool', 'some_int', 'some_list', 'some_str']
+
+    def configure_defaults(self, conf):
+        super().configure_defaults(conf)
+        conf.set('some_bool', True)
+        conf.set('some_int', 1776)
+        conf.set('some_list', ['red', 'white', 'blue'])
+        conf.set('some_str', 'America!')
+        conf.set('some_nonwhitelisted_key', 1492)
+
+
+@genty
 class TestBaseConfigLoader(BaseUnitTestCase):
 
+    @genty_dataset(
+        bool_type=('some_bool', 'False', False),
```

Comment:
The dataset for bool_type is the test that would have failed before the above fix.